plugin wants to execute processes

Tika tries to test if it can run an OCR executable, but does not properly handle SecurityException.

This means this plugin (or its tests) will not work out of box with ES 2.0, unless users disable security completely.

IMO this is a bug in tika ExternalParser.check(), it catches just about every other exception on the planet but doesn't handle the securityexception case:

  > Caused by: java.security.AccessControlException: access denied ("java.io.FilePermission" "<<ALL FILES>>" "execute")
   > 	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:457)
   > 	at java.security.AccessController.checkPermission(AccessController.java:884)
   > 	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
   > 	at java.lang.SecurityManager.checkExec(SecurityManager.java:799)
   > 	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1018)
   > 	at java.lang.Runtime.exec(Runtime.java:620)
   > 	at java.lang.Runtime.exec(Runtime.java:485)
   > 	at org.apache.tika.parser.external.ExternalParser.check(ExternalParser.java:340)
   > 	at org.apache.tika.parser.ocr.TesseractOCRParser.hasTesseract(TesseractOCRParser.java:118)
   > 	at org.apache.tika.parser.ocr.TesseractOCRParser.getSupportedTypes(TesseractOCRParser.java:91)
   > 	at org.apache.tika.parser.CompositeParser.getParsers(CompositeParser.java:95)
   > 	at org.apache.tika.parser.DefaultParser.getParsers(DefaultParser.java:101)
   > 	at org.apache.tika.parser.CompositeParser.getSupportedTypes(CompositeParser.java:254)
   > 	at org.apache.tika.parser.CompositeParser.getParsers(CompositeParser.java:95)
   > 	at org.apache.tika.parser.CompositeParser.getParser(CompositeParser.java:234)
   > 	at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:269)
   > 	at org.apache.tika.parser.AutoDetectParser.parse(AutoDetectParser.java:120)
   > 	at org.apache.tika.Tika.parseToString(Tika.java:506)
   > 	at org.elasticsearch.index.mapper.attachment.AttachmentMapper.parse(AttachmentMapper.java:479)

--- a/pom.xml
+++ b/pom.xml
@@ -33,9 +33,6 @@
 
     <properties>
         <!-- If we need to define any specific property -->
-        <!-- see https://github.com/elastic/elasticsearch-mapper-attachments/issues/129
-             this plugin is currently unusable unless users disable security -->
-        <tests.security.manager>false</tests.security.manager>
     </properties>
 
     <dependencies>
@@ -43,7 +40,7 @@
         <dependency>
             <groupId>org.apache.tika</groupId>
             <artifactId>tika-parsers</artifactId>
-            <version>1.8</version>
+            <version>1.9</version>
             <exclusions>
                 <!-- Not Apache2 License compatible -->
                 <exclusion>

